staging/src/k8s.io/kubectl/pkg/cmd/create/create_service.go

```diff
@@ -57,19 +216,12 @@ var (
     kubectl create service clusterip my-cs --clusterip="None"`))
 )
 
-func addPortFlags(cmd *cobra.Command) {
-	cmd.Flags().StringSlice("tcp", []string{}, "Port pairs can be specified as '<port>:<targetPort>'.")
-}
-
-// ServiceClusterIPOpts holds the options for 'create service clusterip' sub command
-type ServiceClusterIPOpts struct {
-	CreateSubcommandOptions *CreateSubcommandOptions
-}
-
 // NewCmdCreateServiceClusterIP is a command to create a ClusterIP service
 func NewCmdCreateServiceClusterIP(f cmdutil.Factory, ioStreams genericclioptions.IOStreams) *cobra.Command {
-	options := &ServiceClusterIPOpts{
-		CreateSubcommandOptions: NewCreateSubcommandOptions(ioStreams),
+	options := &ServiceOpts{
```

Comment:
I'd prefer a proper `NewServiceOptions` method accepting different type as argument and defaulting the rest.